sktime/classification/early_classification/_teaser.py

```diff
@@ -0,0 +1,563 @@
+
+An early classifier using a one class SVM's to determine decision safety with a
+time series classifier.
+"""
+
+__author__ = ["MatthewMiddlehurst"]
+__all__ = ["TEASER"]
+
+import copy
+
+import numpy as np
+from joblib import Parallel, delayed
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.model_selection import GridSearchCV, cross_val_predict
+from sklearn.svm import OneClassSVM
+from sklearn.utils import check_random_state
+
+from sktime.base._base import _clone_estimator
+from sktime.classification.base import BaseClassifier
+from sktime.classification.dictionary_based import MUSE, WEASEL
+from sktime.classification.feature_based import Catch22Classifier
+
+
+class TEASER(BaseClassifier):
+    """Two-tier Early and Accurate Series Classifier (TEASER).
+
+    An early classifier which uses one class SVM's trained on prediction probabilities
+    to determine whether an early prediction is safe or not.
+
+    Overview:
+        Build n classifiers, where n is the number of classification_points.
+        For each classifier, train a one class svm used to determine prediction safety
+        at that series length.
+        Tune the number of consecutive safe svm predictions required to consider the
+        prediction safe.
+
+        While a prediction is still deemed unsafe:
+            Make a prediction using the series length at classification point i.
+            Decide whether the predcition is safe or not using decide_prediction_safety.
+
+    Parameters
+    ----------
+    estimator: sktime classifier, default=None
+        An sktime estimator to be built at each of the classification_points time
+        stamps. Defaults to a WEASEL classifier.
+    one_class_classifier: one-class sklearn classifier, default=None
+        An sklearn one-class classifier used to determine whether an early decision is
+        safe. Defaults to a tuned one-class SVM classifier.
+    one_class_param_grid: dict or list of dict, default=None
+        The hyper-parameters for the one-class classifier to learn using grid-search.
+        Dictionary with parameters names (`str`) as keys and lists of parameter settings
+        to try as values, or a list of such dictionaries.
+    classification_points : List or None, default=None
+        List of integer time series time stamps to build classifiers and allow
+        predictions at. Early predictions must have a series length that matches a value
+        in the _classification_points List. Duplicate values will be removed, and the
+        full series length will be appeneded if not present.
+        If None, will use 20 thresholds linearly spaces from 0 to the series length.
+    n_jobs : int, default=1
+        The number of jobs to run in parallel for both `fit` and `predict`.
+        ``-1`` means using all processors.
+    random_state : int or None, default=None
+        Seed for random number generation.
+    return_safety_decisions : bool, default=True
```

Comment:
I just noticed, that `score()` has to enforce `return_safety_decisions` to be set to false. Otherwise it will run into an error.

How can that best be done?

Reply:
Also it is unclear to me how to derive the actual earliness ...

Reply:
And how would I get the accuracy and earliness after training? Maybe store it as a local variable in fit? Or does that change the classifier again, leading to an error?

Reply:
It would need a custom score function.

I calculated earliness using state_info and the _classification_points array before. Could add another column containing the input series length so it is easier for others. 

Do you mean the accuracy and earliness from the _fit parameter search? could just store them in i.e. self._fit_accuracy and self._fit_earliness.

Reply:
>  accuracy and earliness from the _fit parameter search?

I wondered about both - i.e., from score and fit :) 

> could just store them in i.e. self._fit_accuracy and self._fit_earliness.

I see - so simply store these values after training in these variables?

> It would need a custom score function.

Could you give pointers or help with the implementation?

Reply:
I can try do the score function. 

I think you would have to return them with hm since its in its own functions, then store the best ones yeah.

Reply:
Cool, thank you! Finally: how might we also get the earliness in score()?

Reply:
New score is basic, could probably edit it to return earliness/hm but at least it works for now

Reply:
I don't know what limitations there are with it really, never edited it. Maybe you can just calculate the earliness from state_info and return it along with accuracy?

Reply:
I have modified the score-method to return acc, earliness and hm. Not sure, if we have to consider the decision-state to compute these values though. Right now, I simply take the last ones as accepted by TEASER.